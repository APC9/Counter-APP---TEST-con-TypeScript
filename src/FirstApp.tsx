
interface Props{
  title: string;
  subTitle?: string;
  name?: string;
}

export const FirstApp = ({ title, subTitle='Sin subtitulo', name='Sin nombre' }:Props) => {
  
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      {/* <p>{ subTitle }</p> */}
      <p>{ subTitle }</p>
      <span>{ name }</span>
    </>
  )
}


