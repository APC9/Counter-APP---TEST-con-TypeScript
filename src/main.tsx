import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp'

import './styles.css'


{/* <FirstApp title={'Hola, Soy Goku'}/> */}
ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  //<React.StrictMode>
    <CounterApp value={100}/>
  //</React.StrictMode>,
)
