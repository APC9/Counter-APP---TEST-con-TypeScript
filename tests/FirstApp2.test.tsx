import { render, screen } from '@testing-library/react'
import React from 'react';
import { FirstApp } from '../src/FirstApp';

describe('Test <FirstApp /> component', ()=>{

  const title='Hola, soy Alberto'
  const subTitle='Soy web developer React'

  test('Debe de hacer Match con el Snapshot', () => {
    
    const {container}=render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  }) 

  test('El titulo debe existir', () => { 
    
    render(<FirstApp title={title}/>);
    expect( screen.getByText(title) ).toBeTruthy()
  })

  test('debe de mostrar el titulo en un H1', () => { 
    
    render(<FirstApp title={title}/>);
    expect( screen.getByRole('heading', {level: 1} ).innerHTML ).toContain(title)
  })

  test('debe de mostrar el subTitulo enviado por Props', () => { 

    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getByText(subTitle) ).toBeTruthy()
  })

})