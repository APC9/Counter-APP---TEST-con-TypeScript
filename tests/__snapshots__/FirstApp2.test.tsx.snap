// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp /> component Debe de hacer Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Alberto
     
  </h1>
  <p>
    Sin subtitulo
  </p>
  <span>
    Sin nombre
  </span>
</div>
`;
